incubator/hnc/internal/validators/object_test.go

```diff
@@ -308,6 +308,34 @@ func TestUserChanges(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "Deny creation of object with invalid select annotation",
+		fail: true,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationSelector: "$foo" + api.LabelTreeDepthSuffix,
```

Comment:
I don't think you _need_ to add the suffix here - `$foo` on its own is definitely not a legal label key :)

Reply:
Yes, you are right. I was trying to simulate the users' scenario more closely, because we are always expecting the user to input the full string. Do you think it's better without this suffix?

Reply:
Since `treeSelector` works so well for the HNC tree labels, I'd actually expect that people using `selector` would be searching for some _other_ kind of string, so I don't think it's necessary to check the HNC one.

But more to the point, our goal here isn't really to test the K8s label selector parsing code - it's that we do something sensible with the error that gets returned. As a result, making the test as simple as possible has a large benefit. Imagine being the next person to modify this test or debug a failure - if you see that input is `foo.tree.hnc.x-k8s.io/depth`, you might spend time looking into whether there's something about _that string_ that's causing a problem because "obviously" that complicated string must be what the test is testing, right? (Wrong!) But if the input string is just "foo" it's much easier to see that there's nothing special about the string.

This is what we mean when we say that most tests should be "minimal" - find the smallest, simplest input that could replicate a problem (or solution) so that if/when the test breaks, it's easier to debug.

Reply:
I see, agreed! Done